code/b-parasite/src/prst/adc.c

```diff
@@ -118,5 +127,25 @@ prst_adc_soil_moisture_t prst_adc_soil_read(double battery_voltage) {
                " %% (percentage); %u (relative)",
                ret.raw, NRF_LOG_FLOAT(percentage * 100), ret.relative);
 #endif
+  return ret;
+}
+
+static inline double get_lux_level(double battery_voltage, nrf_saadc_value_t raw_adc_output) {
```

Comment:
This function should be similar to the `prst_adc_batt_read` above. We need to:
- Sample the `PRST_ADC_PHOTO_CHANNEL` by calling `sample_adc_channel`. Let's call its output `raw_photo_output`
- Transform that `raw_photo_output` (a 10-bit integer) into an analog, floating point voltage by doing `photo_voltage = (3.6 * raw_photo_output) / (1 << PRST_ADC_RESOLUTION)`
- This sampled value will be the voltage in the middle of the voltage divider formed by the photoresistor and R10:

```
GND -----(PR == photo resistor)----- photo_voltage ------(R10 == 10k Ohm)------- (battery_voltage)
```
Now we have everything we need to estimate the value in Ohms for the photo resistor 👌

Reply:
Done.